components/coins-status/index.js

```diff
@@ -1,43 +1,110 @@
-import { useEffect } from 'react';
+import { useState, useEffect } from 'react';
 import { useAnimateValue } from './useCoinStatus';
 import styles from './coin.module.css';
 import PropTypes from 'prop-types';
 
+const BASE_API_URL = process.env.NEXT_PUBLIC_BASE_API_URL;
+const WALLET_URL = `${BASE_API_URL}/wallet`;
+const CURRENCIES = {
+  NEELAM: 'neelam',
+  DINERO: 'dinero',
+};
+
 const Coins = (props) => {
-  const [coins, changeCoins] = useAnimateValue(0, props.coin.value, 5000);
+  const { coin, balance, animate } = props;
+  const [animatedBalance, changeAnimatedBalance] = useAnimateValue(
+    0,
+    balance,
+    5000
+  );
+  const currencyColor =
+    coin === CURRENCIES.NEELAM
+      ? styles.blue
+      : coin === CURRENCIES.DINERO
+      ? styles.green
+      : styles.gray;
+
   useEffect(() => {
-    changeCoins(props.coin.value);
-  }, [props.coin.value, changeCoins]);
+    changeAnimatedBalance(balance);
+  }, [balance, changeAnimatedBalance]);
+
   return (
     <div className={styles.cointypeIndicator}>
       <div className={styles.coinData}>
-        <p> {props.coin.name}</p>
-        <p> {coins} </p>
+        <p>{coin}</p>
+        <p>{animate ? animatedBalance : balance}</p>
       </div>
       <div className={`coin ${styles.rotateVertCenter}`}></div>
       <style jsx>{`
         .coin {
-          background-color: ${props.coin.color};
-          width: calc(4.25em / 2);
-          height: calc(4.25em / 2);
+          background-color: ${currencyColor};
+          width: calc(5em / 2);
+          height: calc(5em / 2);
           border-radius: 50%;
           box-shadow: 3px 3px 3px #000;
-          border: 1px solid ${props.coin.borderColor};
+          border: 1px solid ${currencyColor};
         }
       `}</style>
     </div>
   );
 };
 
-const CoinsStatus = (props) => {
-  let coinsArray = props.coins.map((coin, index) => {
-    return <Coins coin={coin} key={index} />;
-  });
-  return <div className={styles.coinsContainer}>{coinsArray}</div>;
+const CoinsStatus = () => {
+  const [currencies, setCurrencies] = useState([]);
+  const [isLoading, setIsLoading] = useState(true);
+  const [isError, setIsError] = useState(false);
+  const coins = [];
+
+  async function fetchData() {
+    await fetch(WALLET_URL, { credentials: 'include' })
+      .then((response) => {
+        if (response.status >= 400 && response.status < 600) {
+          throw new Error('Bad response from server');
+        }
+        return response.json();
+      })
+      .then((ResponseJson) => {
```

Comment:
1. Can we just stick to arrow functions? (Since we've already used them everywhere)
```JS
const fetchData => async () {
....
}
```

2. If we are already using async on the top, why use `.then()` in that case? Let's use `await` instead

3. `401` isn't bad response from the server, right? The user is not logged in, nothing broken from server-side.

4. Can we have a special message in case of `401`? (When the user is not logged in)

Like `Please login to view your balance`? And have the login link (https://github.com/login/oauth/authorize?client_id=23c78f66ab7964e5ef97) attached to the `login` text?

Reply:
1. Converted into arrow function
2. Need discussion.
3. That was generic error message. Refer 4.
4. Specific error message provided as suggested with login link